### PR TITLE
Fix method to enable in-home doorbell chime

### PR DESCRIPTION
**Pull request by briangoldstein**
I am currently working on adding code to the Home Assistant ring core component to support `async_set_existing_doorbell_type` and `async_set_existing_doorbell_type_enabled`. In working on the switch to enable/disable the in-home chime, I found that Ring is expecting a string, not a boolean value. Error below:

> ring_doorbell.exceptions.RingError: Unknown error during query of url https://api.ring.com/clients_api/doorbots/<doorbell id removed>: Invalid variable type: value should be str, int or float, got True of type <class 'bool'>

Ring is expecting 0 or 1 when enabling and disabling.  I have tested the code change with a modified test.py with the following body between `await ring.async_update_data()` and `await auth.async_close()`

```
    # Fetch all available doorbells
    devices = ring.devices()
    doorbells=devices['doorbots']

    # Flip the status of the in-home chime for each doorbell
    for doorbell in list(doorbells):
        await doorbell.async_update_health_data()
        print()
        print()
        print('Address:    %s' % doorbell.address)
        print('Family:     %s' % doorbell.family)
        print('ID:         %s' % doorbell.id)
        print('Name:       %s' % doorbell.name)
        print('In-Home Chime Type: %s' % doorbell.existing_doorbell_type)
        print('In-Home Chime Status: %s' % doorbell.existing_doorbell_type_enabled)
        print()
        print()
        if doorbell.existing_doorbell_type_enabled:
            print('Turning Off In-Home Chime')
            await doorbell.async_set_existing_doorbell_type_enabled(0)
            time.sleep(5)  # Pause for 5 seconds
        else:
            print('Turning On In-Home Chime')
            await doorbell.async_set_existing_doorbell_type_enabled(1)
            time.sleep(5)  # Pause for 5 seconds
        print()
        print()
        print('In-Home Chime Type: %s' % doorbell.existing_doorbell_type)
        print('In-Home Chime Status: %s' % doorbell.existing_doorbell_type_enabled)

```

### CODE REVIEW

File: ring_doorbell/doorbot.py [hunk 1]
```diff
@@ -214,14 +213,17 @@ def existing_doorbell_type_enabled(self) -> bool | None:
...
         return False
 
-    async def async_set_existing_doorbell_type_enabled(self, value: bool) -> None:  # noqa: FBT001
-        """Enable/disable the existing doorbell if Digital/Mechanical."""
+    async def async_set_existing_doorbell_type_enabled(self, value: int) -> None:
+        """Enable/disable the existing doorbell if Digital/Mechanical.
+        0: Off
+        1: On
+        """
+
```
Comment (sdb9696) [inline ring_doorbell/doorbot.py]:
```suggestion
    async def async_set_existing_doorbell_type_enabled(self, enable: bool) -> None: # noqa: FBT001
        """Enable/disable the existing doorbell if Digital/Mechanical."""
	value = 1 if enable else 0
```
We should maintain the `bool` `in the public interface and convert it internally to the value the api expects.

**Review by sdb9696 — CHANGES_REQUESTED** [review]
As per comment please keep the public api unchanged.

Also please ensure you've done `poetry install` and run `pre-commit` to ensure the CI checks pass locally.

File: ring_doorbell/doorbot.py (hunk 1 above)
Comment (briangoldstein) [reply ring_doorbell/doorbot.py]: Understood and thank you for the quick response! Will get to work on those changes and make sure to get poetry in place.

Comment (briangoldstein) [reply ring_doorbell/doorbot.py]: Thanks! Took this one through the proper steps. I also validated the change against my test script, this time using `await doorbell.async_set_existing_doorbell_type_enabled(False)` and `await doorbell.async_set_existing_doorbell_type_enabled(True)` to flip back and forth and it worked as expected, including adding the bool check logic back in,

**Comment by sdb9696** [comment]
I've never actually used these methods. What do they actually do? Can you actually set a doorbell type to manual or digital.  Is this something available on all devices that derive from doorbots? i.e. stickup_cams etc.

**Comment by briangoldstein** [comment]
It's only available for (most) doorbell models.  It lets the Ring doorbell trigger the in-home doorbell chime.

Not present = you don't have one (or want to use it)
Mechanical = the traditional ding dong physical bell
Digital = the fancy digital versions of doorbell chimes

For digital, you can also set the duration, but I do not have a digital in-home chime so I can't test that aspect and will leave it out of the home assistant changes I will submit.

I used to use a homebridge automation that would turn the in-home doorbell off during the day to prevent the dogs from going nuts, but then would have it turn on at night when our phones are on do not disturb or if we had someone watching the house.

https://ring.com/support/articles/vk1ml/configuring-your-doorbell-for-an-in-home-chime

**Comment by sdb9696** [comment]
Ah ok that makes sense, very helpful thanks.  So the doorbell doesn't actually know what kind of chime it's connected to? It has to be told in the app and the api?

**Comment by briangoldstein** [comment]
You're welcome! That is correct, this is what it looks like in the app when you set mechanical, digital, not present. I'll make sure to include some of this information when I submit the HA ring PR as well. 

IMG_2380
IMG_2381

**Comment by sdb9696** [comment]
Cool. So I think it would be good to have test coverage for this, especially if it's going to be enabled in HA.

**Comment by briangoldstein** [comment]
Sounds good. I am brand new to pytest, so will read up on it this weekend, but may take some time to implement.

**Comment by briangoldstein** [comment]
@sdb9696 I've added a test for the async_set_existing_doorbell_type_enabled function. Please let me know if you had additional tests in mind, thanks!

**Review by sdb9696 — APPROVED** [review]

**Comment by sdb9696** [comment]
So that's now been [released](https://github.com/python-ring-doorbell/python-ring-doorbell/releases/tag/0.9.3) and should be bumped in HA once [HA 125087](https://github.com/home-assistant/core/pull/125087) is merged.  Feel free to hmu on discord (sdb9696) if you want to chat about how best to implement support for this in HA.

**Pull request merged**